cmd/kuke/init: KUKE_INIT_KUKEOND_IMAGE env var silently ignored — resolveKukeondImage bypasses viper binding

## Repro

Worktree: `main` at 72cddbe (\`fix: pre-flight containerd reachability in kuke init\`).

Add the following test to \`cmd/kuke/init/\` and run \`go test ./cmd/kuke/init/... -run TestReproduceKukeondImageEnvIgnored -v\`:

\`\`\`go
package init_test

import (
    "os"
    "testing"

    "github.com/eminwux/kukeon/cmd/config"
    "github.com/spf13/viper"
)

func TestReproduceKukeondImageEnvIgnored(t *testing.T) {
    viper.Reset()
    t.Cleanup(viper.Reset)
    origVersion := config.Version
    origRepo := config.KukeondImageRepo
    t.Cleanup(func() {
        config.Version = origVersion
        config.KukeondImageRepo = origRepo
    })
    config.Version = "v1.0.0"
    config.KukeondImageRepo = "ghcr.io/eminwux/kukeon"

    os.Setenv("KUKE_INIT_KUKEOND_IMAGE", "docker.io/library/kukeon-local:dev")
    t.Cleanup(func() { _ = os.Unsetenv("KUKE_INIT_KUKEOND_IMAGE") })

    got := ResolveKukeondImage()
    if got != "docker.io/library/kukeon-local:dev" {
        t.Errorf("env value ignored: got %q, want %q", got, "docker.io/library/kukeon-local:dev")
    }
}
\`\`\`

(\`ResolveKukeondImage\` already exists as a test-only export in \`cmd/kuke/init/init_test.go\`.)

## Observed

\`\`\`
=== RUN   TestReproduceKukeondImageEnvIgnored
    reproducer_test.go:31: env value ignored: got "ghcr.io/eminwux/kukeon:v1.0.0", want "docker.io/library/kukeon-local:dev"
--- FAIL: TestReproduceKukeondImageEnvIgnored (0.00s)
\`\`\`

The operator-supplied env value \`docker.io/library/kukeon-local:dev\` is silently dropped; \`resolveKukeondImage\` returns the computed default (\`KukeondImageRepo:Version\`) instead.

In a real \`kuke init\` invocation the symptom is: \`KUKE_INIT_KUKEOND_IMAGE=… sudo kuke init\` (without \`--kukeond-image\` and without a \`ServerConfiguration\` setting \`kukeondImage\`) provisions the kukeond cell with the compiled-in default registry image, not the value the operator exported.

## Root cause

\`KUKE_INIT_KUKEOND_IMAGE\` is declared with \`viper.BindPFlag\` against \`--kukeond-image\` in \`cmd/kuke/init/init.go:115\`, but \`viper.BindEnv\` is **never** called for this key — \`grep -rn 'KUKE_INIT_KUKEOND_IMAGE\.BindEnv' .\` returns no matches anywhere in the tree, and there is no \`viper.AutomaticEnv()\` / \`SetEnvPrefix\` either. \`cmd/kukeond/kukeond.go:179 bindEnvVars()\` covers the daemon-side keys but the \`kuke init\` path has no equivalent.

The asymmetry creates a worst-of-both-worlds failure mode: \`applyServerConfiguration\` in \`cmd/kuke/init/init.go:207\` correctly detects the env var via \`envSet()\` (which uses \`os.LookupEnv\` directly) and skips the \`viper.Set\` of the \`ServerConfiguration\` value — so the env *blocks* the spec from winning — but downstream \`viper.GetString(config.KUKE_INIT_KUKEOND_IMAGE.ViperKey)\` in \`resolveKukeondImage\` (\`cmd/kuke/init/init.go:236\`) returns \`\"\"\` because no env binding exists, so the fallback path runs. Net effect: the env value is recognized as a gate but its content is lost, breaking the documented \`--flag > env > ServerConfiguration > default\` precedence rule.

This is the same bug class as the closed #190 (\`KUKEON_GET_OUTPUT env var silently ignored — ParseOutputFormat bypasses viper\`); the same omission likely affects other \`KUKE_INIT_*\` keys (\`KUKE_INIT_NO_WAIT\`, \`KUKE_INIT_FORCE_REGENERATE_CNI\`, \`KUKE_INIT_SERVER_CONFIGURATION\`, \`KUKE_INIT_REALM\`, \`KUKE_INIT_SPACE\`) — the suggested fix should sweep the lot rather than patching just \`KUKEOND_IMAGE\`.

## Suggested fix

Mirror \`cmd/kukeond/kukeond.go\`'s \`bindEnvVars()\` on the kuke side: add a \`bindEnvVars()\` (or extend \`cmd/kuke/kuke.go\`'s \`loadConfig\`, which already binds the shared \`KUKEON_*\` keys) that iterates every \`KUKE_INIT_*\` \`config.Var\` and calls \`v.BindEnv()\`. Then add an \`init_test.go\` regression that mirrors PR #398's \`TestApplyServerConfigurationEnvOverridesConfig\` *without* the explicit per-test \`BindEnv\` calls — proving the production code path wires env binding for \`KUKE_INIT_KUKEOND_IMAGE\` and \`KUKEON_RUN_PATH\` end-to-end.

Surfaced alongside #186 / PR #398 (test-only mirror).

--- a/cmd/kuke/init/init.go
+++ b/cmd/kuke/init/init.go
@@ -92,7 +92,28 @@ func setupInitCmd(cmd *cobra.Command) error {
 		return fmt.Errorf("failed to set persistent flags: %w", err)
 	}
 
+	bindEnvVars()
 	return nil
+}
+
+// bindEnvVars wires each kuke-init viper key to its `KUKE_INIT_*`
+// environment variable. Without these calls applyServerConfiguration's
+// envSet check skips the YAML write yet viper has no env binding to read
+// the value back from — so the operator-exported value is recognised as
+// a gate but its content is silently dropped, leaving the flag default
+// to win. Mirrors the BindEnv block in cmd/kukeond/kukeond.go's
+// bindEnvVars and the KUKEON_* binds in cmd/kuke/kuke.go's loadConfig.
+func bindEnvVars() {
+	for _, v := range []config.Var{
+		config.KUKE_INIT_REALM,
+		config.KUKE_INIT_SPACE,
+		config.KUKE_INIT_KUKEOND_IMAGE,
+		config.KUKE_INIT_NO_WAIT,
+		config.KUKE_INIT_FORCE_REGENERATE_CNI,
+		config.KUKE_INIT_SERVER_CONFIGURATION,
+	} {
+		_ = v.BindEnv()
+	}
 }
 
 func setFlags(cmd *cobra.Command) error {

--- a/cmd/kuke/init/init_test.go
+++ b/cmd/kuke/init/init_test.go
@@ -28,6 +28,7 @@ import (
 	"unsafe"
 
 	"github.com/eminwux/kukeon/cmd/config"
+	"github.com/eminwux/kukeon/cmd/kuke"
 	initpkg "github.com/eminwux/kukeon/cmd/kuke/init"
 	"github.com/eminwux/kukeon/cmd/types"
 	"github.com/eminwux/kukeon/internal/controller"
@@ -802,30 +803,31 @@ func TestRunInitErrors(t *testing.T) {
 // init-specific knobs (KUKE_INIT_KUKEOND_IMAGE, KUKEON_RUN_PATH) so a future
 // change to applyServerConfiguration in init.go cannot regress the env
 // branch while the kukeond sibling test stays green.
+//
+// Issue #399 regression guard: the env bindings are wired exclusively via
+// the production code path — kuke.LoadConfig() for the shared KUKEON_*
+// keys and NewInitCmd() (which calls init's bindEnvVars()) for the
+// KUKE_INIT_* keys. Removing either call from production drops the
+// env binding for the corresponding key, this test fails, and the
+// silent-env-drop bug is caught before it ships.
 func TestApplyServerConfigurationEnvOverridesConfig(t *testing.T) {
 	t.Cleanup(viper.Reset)
 	viper.Reset()
 
+	// Set env before kuke.LoadConfig so the BindEnv'd lookup returns the
+	// env value and loadConfig's empty-default seeding (viper.Set of
+	// DefaultRunPath when KUKEON_RUN_PATH reads empty) does not stamp
+	// the runtime default on top of the env binding.
+	t.Setenv(config.KUKE_INIT_KUKEOND_IMAGE.EnvVar(), "ghcr.io/eminwux/kukeon:from-env")
+	t.Setenv(config.KUKEON_ROOT_RUN_PATH.EnvVar(), "/opt/kukeon-from-env")
+
+	if err := kuke.LoadConfig(); err != nil {
+		t.Fatalf("kuke.LoadConfig: %v", err)
+	}
 	cmd := initpkg.NewInitCmd()
 	if cmd == nil {
 		t.Fatal("NewInitCmd() returned nil")
 	}
-
-	// NewKukeondCmd wires its env bindings via bindEnvVars(); NewInitCmd
-	// defers env binding to kuke.go's loadConfig, which only registers the
-	// shared keys. Bind the init-specific keys explicitly so viper.Get
-	// reflects the env value once t.Setenv fires.
-	if err := config.KUKE_INIT_KUKEOND_IMAGE.BindEnv(); err != nil {
-		t.Fatalf("BindEnv KUKE_INIT_KUKEOND_IMAGE: %v", err)
-	}
-	if err := config.KUKEON_ROOT_RUN_PATH.BindEnv(); err != nil {
-		t.Fatalf("BindEnv KUKEON_ROOT_RUN_PATH: %v", err)
-	}
-
-	// Operator exported KUKE_INIT_KUKEOND_IMAGE and KUKEON_RUN_PATH; the
-	// on-disk ServerConfiguration must not clobber them.
-	t.Setenv(config.KUKE_INIT_KUKEOND_IMAGE.EnvVar(), "ghcr.io/eminwux/kukeon:from-env")
-	t.Setenv(config.KUKEON_ROOT_RUN_PATH.EnvVar(), "/opt/kukeon-from-env")
 
 	spec := v1beta1.ServerConfigurationSpec{
 		KukeondImage: "ghcr.io/eminwux/kukeon:from-config",

